test/runtest.py

```diff
@@ -1543,6 +1543,36 @@ def f():
       time.sleep(0.1)
 
     self.assertEqual(found_message, True)
+
+    ray.worker.cleanup()
+
+  def testTaskProfileAPI(self):
+    ray.init(redirect_output=True)
+
+    @ray.remote
+    def f():
+      return "output"
+
+    f.remote()
+
+    start_time = time.time()
+    while time.time() - start_time < 10:
+      profiles, events = ray.global_state.task_profiles()
+      for profile in profiles:
```

Comment:
One potential problem with this test is that if `profiles` is empty (which it will be at the very beginning), then the test doesn't check anything. We should also make sure that `profiles` has exactly one element.